### DISCUSSION TITLE
Add Text Node

### DISCUSSION THREAD

**Discussion by Doppelklick** [CariusLars/ar_flutter_plugin#25] [Q&A] [answered]
Hi,
thx for the plugin :-) 

I'm very new at AR and the examples are a good start. But now I want to add a dynamic text (distance from camera to object). 

I didn't find examples - so can you give me an idea how to do this? 

Best,
Dom

**Comment by CariusLars** [comment, marked as answer]
Hi @Doppelklick,

I'm not 100% sure what you mean with dynamic text, but here are some ideas:
- If you want to "measure" the distance from the camera to the object and display that as a text in your app, you can use the `getCameraPose()` in the [ARSessionManager](https://github.com/CariusLars/ar_flutter_plugin/blob/feature/gps-objects/lib/managers/ar_session_manager.dart) and calculate the distance between the camera and your object using the returned pose and the world position of your respective object. You can then add a text widget in your flutter app that displays the result, dynamic updates could be achieved by e.g. calling the method in each new frame and updating the widget using stateful widgets or the provider pattern. Please note that the `getCameraPose()` method is not on the main branch yet, let me know if you need it and I can review the code and merge it into main.
- If you want to display text in your 3D-scene, you could create a new [nodetype](https://github.com/CariusLars/ar_flutter_plugin/blob/main/lib/datatypes/node_types.dart), add the text in the [Node's](https://github.com/CariusLars/ar_flutter_plugin/blob/main/lib/models/ar_node.dart) `data` property and implement platform-specific handlers that create a node object from your text at the desired postion

Hope that helps, let me know if you need more information! Contributions to the plugin are always very welcome in case you want to add additional functionality to solve your problem :)

Cheers

Lars
